src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java

```diff
@@ -125,32 +129,45 @@ private List<PublicationStatusError> buildErrorListFromPublicationTenantRequestE
+    List<Future<PublicationTenantRequestEntity>> futures = new ArrayList<>();
+
+    ExecutorService executor = Executors.newFixedThreadPool(MAX_ACTIVE_THREADS);
 
     for (String tenantId : publicationRequest.getTenants()) {
       try {
         PublicationTenantRequestEntity ptrEntity = buildPublicationRequestEntity(publicationRequest, createdPublicationEntity, tenantId);
-
         var savedPublicationTenantRequest = savePublicationTenantRequest(ptrEntity, folioExecutionContext);
-
-        semaphore.acquire();
-        var future = executeAsyncHttpRequest(publicationRequest, tenantId, folioExecutionContext)
-          .whenComplete((ok, err) -> semaphore.release())
-          .handle((response, t) -> updatePublicationTenantRequest(response, t, savedPublicationTenantRequest, folioExecutionContext));
+        var future = executor.submit(() -> updatePublicationTenantRequest(publicationRequest, tenantId, savedPublicationTenantRequest));
         futures.add(future);
       } catch (RuntimeException | JsonProcessingException e) {
-        log.error("publishRequest:: failed to save publication tenant request", e);
-        semaphore.release();
+        log.error("processTenantRequests:: failed to save publication tenant request", e);
         futures.add(CompletableFuture.failedFuture(e));
-      } catch (InterruptedException ie) {
-        log.error("publishRequest:: failed to acquire semaphore permit", ie);
-        futures.add(CompletableFuture.failedFuture(ie));
-        Thread.currentThread().interrupt();
       }
     }
-    CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]))
-      .thenAccept(cf -> updatePublicationsStatus(futures, createdPublicationEntity, folioExecutionContext));
+
+    executor.shutdown();
+    try {
+      if (executor.awaitTermination(300, TimeUnit.SECONDS)) {
+        updatePublicationsStatus(futures, createdPublicationEntity, folioExecutionContext);
+        executor.shutdownNow();
+      }
+    } catch (InterruptedException ie) {
+      log.error("processTenantRequests:: failed to shutdown executor", ie);
+      Thread.currentThread().interrupt();
+    }
+
+  }
+
+  PublicationTenantRequestEntity updatePublicationTenantRequest(PublicationRequest publicationRequest, String tenantId,
+    PublicationTenantRequestEntity savedPublicationTenantRequest) {
+    PublicationTenantRequestEntity updatedPtre;
+    try {
+      var response = executeHttpRequest(publicationRequest, tenantId, folioExecutionContext);
+      updatedPtre = updateSucceedPublicationTenantRequest(response, savedPublicationTenantRequest, folioExecutionContext);
+    } catch (Exception e) {
+      updatedPtre = updateFailedPublicationTenantRequest(e, savedPublicationTenantRequest, folioExecutionContext);
+    }
+    return updatedPtre;
   }
 
   private PublicationTenantRequestEntity savePublicationTenantRequest(PublicationTenantRequestEntity ptrEntity, FolioExecutionContext centralTenantContext) {
```

Comment:
it looks like we're setting context with same content as it was, could be removed